### ISSUE TITLE
No way to start conversations with Facebook users

### ISSUE THREAD

**Issue by tfreedman** [matrix-hacks/matrix-puppet-facebook#2]
If someone on facebook messages me, I'll be dropped into a room properly on matrix. However, there's no way to get a list of all my friends or anything, if I want to start a conversation with someone who I don't have a currently open window with in matrix.

**Comment by kfatehi** [comment]
That's true. Doing it this way was much easier and minimalist in that only those that message you would cause a room to be created, rather than to create rooms for everyone on your list (or in your contacts book, in the case of iMessage, for example), many of whom do not message you.

That said I think there should be a way to initiate without having to go to the third party client (e.g. facebook.com or iMessage). One way could be through bang commands in a bot room. For example we can do something like this, where to start talking to you I can tell `facebookbot` the following:

`!chat tfreedman`

and then this will cause the bridge code to look for a contact with that name.

if multiple matches, have me select one by entering the number of the correct match (1,2,3,)

if no matches, let me know.

once matched, create the room, create the ghost user in that room, and allow communication..

thoughts? can you think of a better way?

**Comment by kfatehi** [comment]
ping @AndrewJDR

**Comment by tfreedman** [comment]
There's a better way, which is what software like Bitlbee does - it creates a room called Facebook, and drops all of your contacts in there. There's no actual talking in the room, but you can right click a name and start PMing that person.

**Comment by kfatehi** [comment]
That's a great solution. I'd like such a special room for having status messages like https://github.com/AndrewJDR/matrix-puppet-bridge/issues/3 and maybe even configuration type stuff. Most of the logic can be implemented generally in https://github.com/AndrewJDR/matrix-puppet-bridge and then the bridges can be modified to take advantage of it by syncing users from the 3rd party side.

**Comment by kfatehi** [comment]
support for this has been merged into the develop branch for now https://github.com/matrix-hacks/matrix-puppet-bridge/pull/11

**Comment by kfatehi** [comment]
Needs to be tested.

**Comment by babolivier** [comment]
Hey there, I recently installed the bridge from master, and although it created me a room with all of my contacts in there, I can't initiate a conversation with one of them by just PMing them. If I open a direct chat in Riot with one of them, it will invite the Facebook contact in it, who will never join, and no message will pass through the bridge. The only log I can see from when I tell Riot to create the direct chat room to

**Comment by JeffAlyanak** [comment]
> 
> 
> Hey there, I recently installed the bridge from master, and although it created me a room with all of my contacts in there, I can't initiate a conversation with one of them by just PMing them. If I open a direct chat in Riot with one of them, it will invite the Facebook contact in it, who will never join, and no message will pass through the bridge. The only log I can see from when I tell Riot to create the direct chat room is
> 
> ```
> EventTimelineSet.addLiveEvent: ignoring duplicate event [...]
> ```
> 
> which happens at the room creation.

I can confirm identical behaviour.

Have you found any workarounds for that?

**Comment by Eluminae** [comment]
Got exactly the same thing here.

**Comment by Eluminae** [comment]
And the room created when someone initiate the conversation looks like "Unnamed Room (#facebook_1000071XXXXXXX:synapse.XXXX.XX)"

Is this normal ?

**Comment by MartenBE** [comment]
Is there any update on this?
